core-ui/src/components/Predefined/Create/OAuthClients/OAuthClients.create.js

```diff
@@ -1,124 +1,177 @@
+        validate={val => val?.length}
+        label={t('oauth2-clients.labels.response-types')}
+        input={Inputs.Checkboxes}
+        options={['id_token', 'code', 'token'].map(type => ({
+          key: type,
+          text: t(`oauth2-clients.response-types.${type}`),
+        }))}
+      />
+      <ResourceForm.FormField
+        required
+        propertyPath="$.spec.grantTypes"
+        validate={val => val?.length}
+        label={t('oauth2-clients.labels.grant-types')}
+        input={Inputs.Checkboxes}
+        options={[
+          'client_credentials',
+          'authorization_code',
+          'implicit',
+          'refresh_token',
+        ].map(type => ({
+          key: type,
+          text: t(`oauth2-clients.grant-types.${type}`),
+        }))}
+      />
+      <ResourceForm.FormField
+        advanced
+        propertyPath="$.spec.clientName"
+        label={t('oauth2-clients.labels.client-name')}
+        tooltipContent={t('oauth2-clients.tooltips.client-name')}
+        input={Inputs.Text}
+      />
+      <ResourceForm.FormField
+        advanced
+        propertyPath="$.spec.secretName"
+        label={t('oauth2-clients.labels.secret-name')}
+        tooltipContent={t('oauth2-clients.tooltips.secret-name')}
+        input={({ value, setValue }) => (
+          <K8sResourceSelectWithUseGetList
+            compact
+            required
+            value={value}
+            resourceType={t('oauth2-clients.labels.secret')}
+            onSelect={setValue}
+            url={`/api/v1/namespaces/${namespace}/secrets`}
+          />
+        )}
+      />
+      <ResourceForm.FormField
+        advanced
+        propertyPath="$.spec.tokenEndpointAuthMethod"
+        label={t('oauth2-clients.labels.auth-method')}
+        input={Inputs.Dropdown}
+        options={[
+          {
+            key: 'none',
+            text: t('oauth2-clients.auth-methods.none'),
+          },
+          {
+            key: 'client_secret_basic',
+            text: t('oauth2-clients.auth-methods.client_secret_basic'),
+          },
+          {
+            key: 'client_secret_post',
+            text: t('oauth2-clients.auth-methods.client_secret_post'),
+          },
+          {
+            key: 'private_key_jwt',
+            text: t('oauth2-clients.auth-methods.private_key_jwt'),
+          },
+        ]}
+      />
+      <TextArrayInput
+        required
+        propertyPath="$.spec.scope"
+        validate={val => !!val}
+        title={t('oauth2-clients.labels.scope')}
+        toInternal={value => value?.split(/ +/) || []}
+        toExternal={value => value.filter(Boolean).join(' ')}
+      />
+      <TextArrayInput
+        advanced
+        propertyPath="$.spec.redirectUris"
+        title={t('oauth2-clients.labels.redirect-uris')}
+      />
+      <TextArrayInput
+        advanced
+        propertyPath="$.spec.postLogoutRedirectUris"
+        title={t('oauth2-clients.labels.post-logout-redirect-uris')}
```

Comment:
Should we use here `type=url` or a `pattern` for input?

Failed to create the OAuth2 Client: OAuth2Client.hydra.ory.sh "aa" is invalid: 
* spec.postLogoutRedirectUris: Invalid value: "b": spec.postLogoutRedirectUris in body should match '\w+:/?/?[^\s]+', 
* spec.redirectUris: Invalid value: "a": spec.redirectUris in body should match '\w+:/?/?[^\s]+'].